src/helpers/cache.js

```diff
@@ -4,54 +4,93 @@
 const config = require('../config');
 const redis = require('redis');
 const bluebird = require('bluebird'); bluebird.promisifyAll(redis.RedisClient.prototype);
-let cacheInstance = null;
-
-// Singleton cache class to get Redis client.
-class Cache {
-  constructor() {
-    if (!cacheInstance) { 
-      if (!config.get('REDIS_HOST')) { throw new Error('You need to define REDIS_HOST environment variable!'); }      
-      this.client = redis.createClient(6379, config.get('REDIS_HOST')); 
-      cacheInstance = this;
-    }    
-    return cacheInstance;
-  }
+const logger = require('../logger').getLogger(module);
+
+const HELPER_NAME = 'Redis cache helper';
+const REDIS_HOST = config.get('REDIS_HOST');
+const REDIS_PORT = config.get('REDIS_PORT');
+const REDIS_RECONNECT_AFTER_FAIL_MS = config.get('REDIS_RECONNECT_AFTER_FAIL_MS') || 60 * 1000;
+const REDIS_CONNECT_TIMEOUT_MS = config.get('REDIS_CONNECT_TIMEOUT_MS') || 2 * 1000;
+
+let isRedisConnected = false;
+
+if (!REDIS_HOST || !REDIS_PORT) {
```

Comment:
Lets put this check and redis client creation back into singleton so it will run only for consumers.